### PR TITLE
JS-3 week1 assignment

### PR DESCRIPTION
**Pull request by daxiousm**

### CODE REVIEW

File: javascript/javascript3/week1/weather/index.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Fetch Weather API</title>
+  </head>
+
+  <body>
+    <h1>Weather Forecast</h1>
+    <input type="text" id="userinput" placeholder="Enter city name" />
+    <button id="button1">SEARCH</button>
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/index.html:13]: button type should be submit.

File: javascript/javascript3/week1/weather/weather.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const button = document.getElementById("button1");
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/weather.js:1]: Try to give better variable name like btnSubmit.

File: javascript/javascript3/week1/weather/weather.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const input = document.getElementById("userinput");
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/weather.js:2]: variable name can be better like : city?

File: javascript/javascript3/week1/weather/weather.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+let body = document.querySelector("body");
+let ul = document.querySelector("ul");
+
+//const apiKey = "c65dee63b1543ccd79345b5fdaa81ee6";
+
+const url = `https://api.openweathermap.org/data/2.5/weather?q=copenhagen&appid=c65dee63b1543ccd79345b5fdaa81ee6&units=metric`;
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/weather.js:9]: what is the benefit of taking user input if you are hardcoding city here = copenhagen, Make it dynamic.

File: javascript/javascript3/week1/weather/weather.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+function inputLength() {
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/weather.js:11]: what is the use of this function ?

File: javascript/javascript3/week1/weather/weather.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+function createListElements() {
+  fetch(url)
+    .then((response) => response.json())
+    .then((data) => {
+      let h2 = document.createElement("h2");
+      h2.appendChild(document.createTextNode(data.name + "_" + "Weather"));
+      body.appendChild(h2);
+      const temp = document.getElementById("temperature");
+      temp.appendChild(
+        document.createTextNode(Math.floor(data.main.temp) + "-" + "kelvin")
```
Comment (piyushpalod) [inline javascript/javascript3/week1/weather/weather.js:23]: what can go wrong here ? if tmp will be null. check for empty or null value.

File: javascript/javascript3/week1/weather/index.html (hunk 1 above)
Comment (daxiousm) [reply javascript/javascript3/week1/weather/index.html:13]: could you explain why it has to be a submit

File: javascript/javascript3/week1/weather/weather.js (hunk 6 above)
Comment (daxiousm) [reply javascript/javascript3/week1/weather/weather.js:23]: i don't understand, we have an input for the city name, how could i make the temp value null. but without a value in a search bar , nothing will be displayed.

File: javascript/javascript3/week1/weather/weather.js (hunk 5 above)
Comment (daxiousm) [reply javascript/javascript3/week1/weather/weather.js:11]: it's a function that checks the input.value.length

**Pull request merged**